k2/python/k2/dense_fsa.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c)  2020  Mobvoi Inc.        (authors: Fangjun Kuang)
+
+import _k2
+import numpy as np
+import torch
+
+
+def dense_fsa(log_probs: torch.Tensor,
+              input_lengths: torch.Tensor) -> _k2.DenseFsaVec:
+    '''Construct a DenseFsa from neural net log-softmax outputs.
+
+    Args:
+      log_probs:
+        A 3-D tensor of dtype ``torch.float32`` with shape ``(N, T, C)``, where
+        ``N`` is the batch size, ``T`` the maximum input length, and ``C`` the
+        number of output classes.
+      input_lengths:
+        A 1-D tensor of dtype ``torch.int32`` with size ``(N)``, where ``N`` is
+        the batch size. It represents the length of the input (must each NOT
+        be greater than ``T``).
+    Returns:
+      An instance of ``_k2.DenseFsaVec``. You can treat it as an opaque object
+      in Python.
+    '''
+    assert log_probs.ndim == 3
+    assert log_probs.dtype == torch.float32
+    assert input_lengths.ndim == 1
+    assert input_lengths.dtype == torch.int32
+
+    N = log_probs.size(0)
+    assert input_lengths.size(0) == N
+
+    num_rows = input_lengths.sum().item() + N
```

Comment:
Is `+N` here for the last row? (`+1` for every one?), maybe it's better to add comments here.

Reply:
The length of every sequence is incremented.
https://github.com/k2-fsa/k2/blob/9b679bdbfd803fdd1e6fc4c5813f251a2b8ebf19/k2/csrc/fsa.h#L118-L120

I'll add more comments for the Python code.

Reply:
Yeah, as Python users may not check C++ code. Thanks!

Reply:
C++ comments are copied to Python.